### PR TITLE
Diagnostics docs re #204

### PR DESCRIPTION
**Pull request by stevemullerworth**
New docs to describe how to add diagnostics to LFRic atmosphere.

### CODE REVIEW

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst [hunk 1]
```diff
@@ -1,11 +1,409 @@
...
++===================================+========================================+
+|  :math:`\mathbb{W}_{3}`           |  ``grid_ref="half_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{theta}`       |  ``grid_ref="full_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{2v}`          |  ``grid_ref="full_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{2h}`          |  ``grid_ref="half_level_edge_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{theta}`       |  ``grid_ref="full_level_face_grid"``   |
```
Comment (james-bruten-mo) [inline source/WorkingPractices/Diagnostics/lfric_diagnostics.rst:189]: Should there be multiple W_theta entries here?

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst [hunk 2]
```diff
@@ -1,11 +1,409 @@
...
+                                         grid_canopy_water,      &
+                                         throughfall,            &
+                                         grid_throughfall)
+
+The ``initialise_diags_for_jules_soil`` procedure calls the LFRic
+atmosphere ``init_diag`` function:
+
+.. code-block:: fortran
+
+    soil_moisture_content_flag = init_diag(soil_moisture_content, &
```
Comment (james-bruten-mo) [inline source/WorkingPractices/Diagnostics/lfric_diagnostics.rst:247]: Being picky, it would be better to align the continuation lines

**Review by james-bruten-mo — CHANGES_REQUESTED** [review]
Just a couple of minor comments

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst (hunk 1 above)
Comment (stevemullerworth) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst:189]: No. One removed.

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst (hunk 2 above)
Comment (stevemullerworth) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst]: Changed all the alignment of these lines.

**Comment by stevemullerworth** [comment]
Fixed the requested issues. Ready for review again.

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst (hunk 1 above)
Comment (james-bruten-mo) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst:189]: Think there were 3 before, now 2 L183 + L189?

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst (hunk 2 above)
Comment (james-bruten-mo) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst]: Cheers

File: source/WorkingPractices/Diagnostics/lfric_diagnostics.rst (hunk 1 above)
Comment (stevemullerworth) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst]: Eek! Now fixed and checked more carefully.

Comment (james-bruten-mo) [reply source/WorkingPractices/Diagnostics/lfric_diagnostics.rst]: Easily done, thanks.

**Review by james-bruten-mo — APPROVED** [review]

**Pull request merged**